File: sample-java/src/main/java/social/bigbone/sample/GetDirectoryAccounts.java
package social.bigbone.sample;

import social.bigbone.MastodonClient;
import social.bigbone.api.entity.Account;
import social.bigbone.api.exception.BigBoneRequestException;
import social.bigbone.api.method.DirectoryMethods;

import java.util.Comparator;
import java.util.List;
import java.util.NoSuchElementException;

public class GetDirectoryAccounts {
    public static void main(final String[] args) throws BigBoneRequestException {
        final String instance = args[0];
        final String accessToken = args[1];

        // instantiate client
        final MastodonClient client = new MastodonClient.Builder(instance)
                .accessToken(accessToken)
                .build();

        // get 40 local accounts that were recently active, skipping the first ten
        final List<Account> accounts = client.directories().listAccounts(
                true,
                DirectoryMethods.AccountOrder.ACTIVE,
                10,
                40
        ).execute();

        // do something with the result; here, we find the oldest account still active and output information about it
        final Account account = accounts.stream().min(Comparator.nullsLast(
                Comparator.comparing((Account acct) -> acct.getCreatedAt().mostPreciseInstantOrNull())
        )).orElseThrow(NoSuchElementException::new);

        System.out.println("@" + account.getAcct() + "@" + instance + " has posted "
                + account.getStatusesCount() + " times since " + account.getCreatedAt().mostPreciseInstantOrNull());
    }
}

File: sample-kotlin/src/main/kotlin/social/bigbone/sample/GetDirectoryAccounts.kt
package social.bigbone.sample

import social.bigbone.MastodonClient
import social.bigbone.PrecisionDateTime
import social.bigbone.api.method.DirectoryMethods
import java.time.Instant

object GetDirectoryAccounts {
    @JvmStatic
    fun main(args: Array<String>) {
        val instance = args[0]
        val accessToken = args[1]

        // instantiate client
        val client = MastodonClient.Builder(instance)
            .accessToken(accessToken)
            .build()

        // get 40 local accounts that were recently active, skipping the first ten
        val accounts = client.directories.listAccounts(
            local = true,
            order = DirectoryMethods.AccountOrder.ACTIVE,
            offset = 10,
            limit = 40
        ).execute()

        // do something with the result; here, we find the oldest account still active and output information about it
        val oldestAccount = accounts
            .filter { it.createdAt is PrecisionDateTime.ExactTime || it.createdAt is PrecisionDateTime.StartOfDay }
            .minBy {
                when (it.createdAt) {
                    is PrecisionDateTime.ExactTime -> (it.createdAt as PrecisionDateTime.ExactTime).instant
                    is PrecisionDateTime.StartOfDay -> (it.createdAt as PrecisionDateTime.StartOfDay).instant
                    // Shouldn't happen, but still returning an EPOCH Instant just in case
                    else -> Instant.ofEpochSecond(0)
                }
            }
        oldestAccount.let {
            println("@${it.acct}@$instance posted ${it.statusesCount} times since ${it.createdAt}")
        }
    }
}
